ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/data/PaywallViewModel.kt

```diff
@@ -61,13 +61,14 @@ internal interface PaywallViewModel {
 }
 
 @OptIn(ExperimentalPreviewRevenueCatUIPurchasesAPI::class, ExperimentalPreviewRevenueCatPurchasesAPI::class)
-@Suppress("TooManyFunctions")
+@Suppress("TooManyFunctions", "LongParameterList")
 internal class PaywallViewModelImpl(
     override val resourceProvider: ResourceProvider,
     private val purchases: PurchasesType = PurchasesImpl(),
     private var options: PaywallOptions,
     colorScheme: ColorScheme,
     private var isDarkMode: Boolean,
+    private var shouldDisplayBlock: ((CustomerInfo) -> Boolean)?,
```

Comment:
Note that we don't update this after it's set. This means that if the developer changes between different required entitlement ids/condition blocks/having conditions and not, the paywall will keep using the first one passed (note that the view model can survive the paywall closing when presented as a dialog, unless using the navigation component.

I'm not sure how big of a problem that would be though...

Reply:
Should this be immutable then?